File: ejercicios/c++/ejercicio027.cpp
#include <iostream>

using namespace std;

int main(int argc, char **argv)
{
	int agno;
	int mes;
	int dia;
	bool bandera;
		
	cout << "ingrese el agno: ";
	cin >> agno;
	
	cout << "ingrese el mes: ";
	cin >> mes;
	
	cout << "ingrese el dia: ";
	cin >> dia;
		
	bandera = true;	
	
	if(mes < 1 || mes > 12){
		bandera = false;
	}else if(mes == 1 || mes == 3 || mes == 5 || mes == 7 || mes == 8 || mes == 10 || mes == 12){
		if(dia > 31){
			bandera = false;
		}
	}else if(mes == 2){		
		if(agno % 4 == 0 && (agno % 100 != 0 || agno % 400 == 0)){
			if(dia > 29){
				bandera = false;
			}
		}else{
			if(dia > 28){
				bandera = false;
			}
		}
	}else if(mes == 4 || mes == 6 || mes == 9 || mes == 11){
		if(dia > 30){
			bandera = false;
		}
	}
	if(bandera){
		cout << agno << "-" << mes << "-" << dia << " es valida" << endl;
	}else{
		cout << agno << "-" << mes << "-" << dia << " no es valida" << endl;
	}
	
	return 0;
}

File: ejercicios/python/ejercicio027.py
bandera = True

agno = int(input("ingrese el agno: "))
mes = int(input("ingrese mes: "))
dia = int(input("ingrese el dia: "))
	
if mes < 1 or mes > 12:
	bandera = False
elif mes == 1 or mes == 3 or mes == 5 or mes == 7 or mes == 8 or mes == 10 or mes == 12:
	if dia > 31:
		bandera = False
elif mes == 2:
	if agno % 4 == 0 and (agno % 100 != 0 or agno % 400 == 0):
		if dia > 29:
			bandera = False
	else:
		if dia > 28:
			bandera = False
elif mes == 4 or mes == 6 or mes == 9 or mes == 11:
	if dia > 30:
		bandera = False
	
if bandera:
	print("%i-%i-%i es valida" %(agno,mes,dia))
else:
	print("%i-%i-%i no es valida" %(agno,mes,dia))
